src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -30,76 +31,82 @@
 /** Provides access to the data stored in Datastore. */
 public class Datastore {
 
-  private DatastoreService datastore;
-
-  public Datastore() {
-    datastore = DatastoreServiceFactory.getDatastoreService();
-  }
-
-  /** Stores the Message in Datastore. */
-  public void storeMessage(Message message) {
-    Entity messageEntity = new Entity("Message", message.getId().toString());
-    messageEntity.setProperty("user", message.getUser());
-    messageEntity.setProperty("text", message.getText());
-    messageEntity.setProperty("timestamp", message.getTimestamp());
-
-    datastore.put(messageEntity);
-  }
-
-  /**
-   * Get messages from a certain query
-   * @param query
-   * @return a list of messages from this query 
-   */
-  public List<Message> getQueryMessages(Query query) {
-
-    List<Message> messages = new ArrayList<>();
-
-    PreparedQuery results = datastore.prepare(query);
-
-    for (Entity entity : results.asIterable()) {
-
-      try {
-        String idString = entity.getKey().getName();
-        UUID id = UUID.fromString(idString);
-        String user = (String) entity.getProperty("user");
-        String text = (String) entity.getProperty("text");
-        long timestamp = (long) entity.getProperty("timestamp");
-
-        Message message = new Message(id, user, text, timestamp);
-        messages.add(message);
-      } catch (Exception e) {
-        System.err.println("Error reading message.");
-        System.err.println(entity.toString());
-        e.printStackTrace();
-      }
-    }
-
-    return messages;
-  }
-
-  /**
-   * Gets messages posted by a specific user.
-   *
-   * @return a list of messages posted by the user, or empty list if user has never posted a
-   *     message. List is sorted by time descending.
-   */
-  public List<Message> getMessages(String user) {
-    Query query =
-        new Query("Message")
-            .setFilter(new Query.FilterPredicate("user", FilterOperator.EQUAL, user))
-            .addSort("timestamp", SortDirection.DESCENDING);
-
-    return getQueryMessages(query);
-  }
-
-  /**
-   * Get all messages from all users 
-   */
-  public List<Message> getAllMessages() {
-    Query query = new Query("Message").addSort("timestamp", SortDirection.DESCENDING); 
-
-    return getQueryMessages(query); 
-   
-  }
+	private DatastoreService datastore;
```

Comment:
Is datastore modifiable?
If it will not be modified later on, please mark it as final. It is always safer that way (and preferred)

Reply:
The week 2 tasks required me to modify this file and I believe we will need to modify them more in the future.

Reply:
Yeah I think we might modify datastore later on too